app/src/main/res/layout/activity_forum.xml

```diff
@@ -1,163 +1,166 @@
+                app:layout_constraintTop_toBottomOf="@id/post_edit_text" />
 
-                <ImageView
-                    android:id="@+id/post_use_camera"
-                    android:layout_width="100dp"
-                    android:layout_height="50dp"
-                    app:srcCompat="@android:drawable/ic_menu_camera"
+            <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:orientation="horizontal"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/post_display_image">
+
+                <LinearLayout
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:orientation="vertical">
+
+                    <ImageView
+                        android:id="@+id/post_use_camera"
+                        android:layout_width="100dp"
+                        android:layout_height="50dp"
+                        android:contentDescription="@string/use_camera"
+                        android:onClick="onClick"
+                        app:srcCompat="@android:drawable/ic_menu_camera" />
+
+                    <ImageView
+                        android:id="@+id/post_use_gallery"
+                        android:layout_width="100dp"
+                        android:layout_height="50dp"
+                        android:contentDescription="@string/use_gallery"
+                        android:onClick="onClick"
+                        app:srcCompat="@android:drawable/ic_menu_gallery" />
+
+                </LinearLayout>
+
+                <Button
+                    android:id="@+id/submit_post"
+                    style="@style/Theme.Meili.button"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
                     android:onClick="onClick"
-                    android:contentDescription="@string/use_camera" />
-
-                <ImageView
-                    android:id="@+id/post_use_gallery"
-                    android:layout_width="100dp"
-                    android:layout_height="50dp"
-                    app:srcCompat="@android:drawable/ic_menu_gallery"
+                    android:text="@string/create_post"
+                    android:theme="@style/Theme.Meili.button" />
+
+                <Button
+                    android:id="@+id/cancel_post"
+                    style="@style/Theme.Meili.button"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
                     android:onClick="onClick"
-                    android:contentDescription="@string/use_gallery" />
+                    android:text="@string/cancel"
+                    android:theme="@style/Theme.Meili.button" />
```

Comment:
Cirrus suggests that the cancel button should always be on the left and the submit on the right.